### PR TITLE
IDE demo: Show names of tokens on hover

### PR DESCRIPTION
**Pull request by anka-213**

### CODE REVIEW

File: lsp/Main.hs [hunk 1]
```diff
@@ -1,45 +1,151 @@
...
+
+      --       _ <- registerCapability STextDocumentCodeLens regOpts $ \_req responder -> do
+      --         let cmd = Command "Say hello" "lsp-hello-command" Nothing
+      --             rsp = List [CodeLens (mkRange 0 0 0 100) (Just cmd) Nothing]
+      --         responder (Right rsp)
+      --       pure ()
+      --     Right _ ->
+      --       sendNotification SWindowShowMessage (ShowMessageParams MtInfo "Not turning on code lenses")
+      --     Left err ->
+      --       sendNotification SWindowShowMessage (ShowMessageParams MtError $ "Something went wrong!\n" <> T.pack (show err))
```
Comment (kharus) [inline lsp/Main.hs:37]: This commented-out block I think we can just remove.
I know where to get it :-)

File: lsp/Main.hs [hunk 2]
```diff
@@ -1,45 +1,151 @@
...
+tshow :: Show a => a -> T.Text
+tshow = T.pack . show
+
+matchesPos' :: Int -> Int -> Token -> Bool
+matchesPos' line col (Token (AlexPn _ l c) len _) =
+  line == l -- && col == c
+  && col >= c && col < c + len
+
+elog :: (MonadIO m, Show a) => a -> m ()
+elog = liftIO . hPutStrLn stderr . tshow
```
Comment (kharus) [inline lsp/Main.hs:83]: Is there standard logging infra in Haskell?

File: lsp/Main.hs [hunk 3]
```diff
@@ -1,45 +1,151 @@
...
+                  (Just "lexer") -- source
+                  (T.pack $ msg err)
+                  Nothing -- tags
+                  (Just (J.List []))
+                ]
+      publishDiagnostics 100 nuri Nothing (partitionBySource diags)
+
+      _ <- sendNotification SWindowShowMessage $ ShowMessageParams MtError $ tshow err
+      return Nothing -- TODO: Report this error as a diagnostic
+      -- One problem with making the diagnostic is that alex by default shows errors as a string
```
Comment (kharus) [inline lsp/Main.hs:122]: You changed this in later commits?

File: lsp/Main.hs (hunk 1 above)
Comment (anka-213) [reply lsp/Main.hs:37]: Good point

File: lsp/Main.hs (hunk 2 above)
Comment (anka-213) [reply lsp/Main.hs:83]: Yes, I just wanted to get some output as quickly as possible.
There are some examples in `Reactor.hs`.

File: lsp/Main.hs (hunk 3 above)
Comment (anka-213) [reply lsp/Main.hs:122]: Yes, but it was pretty hacky.

**Comment by anka-213** [comment]
@kharus Thanks for actually giving me a review!

**Review by kharus — APPROVED** [review]

**Pull request merged**